package/src/components/Channel/hooks/useCreateChannelCapabilitiesContext.ts

```diff
@@ -0,0 +1,81 @@
+  DefaultReactionType,
+  DefaultUserType,
+  UnknownType,
+} from '../../../types/types';
+import type { Channel } from 'stream-chat';
+
+export const useCreateChannelCapabilitiesContext = <
+  At extends UnknownType = DefaultAttachmentType,
+  Ch extends UnknownType = DefaultChannelType,
+  Co extends string = DefaultCommandType,
+  Ev extends UnknownType = DefaultEventType,
+  Me extends UnknownType = DefaultMessageType,
+  Re extends UnknownType = DefaultReactionType,
+  Us extends UnknownType = DefaultUserType,
+>({
+  channel,
+  mutesEnabledProp,
+  pinMessageEnabledProp,
+  quotedRepliesEnabledProp,
+  reactionsEnabledProp,
+  readEventsEnabledProp,
+  threadRepliesEnabledProp,
+  typingEventsEnabledProp,
+  uploadsEnabledProp,
+}: {
+  channel: Channel<At, Ch, Co, Ev, Me, Re, Us>;
+  mutesEnabledProp?: boolean;
+  pinMessageEnabledProp?: boolean;
+  quotedRepliesEnabledProp?: boolean;
+  reactionsEnabledProp?: boolean;
+  readEventsEnabledProp?: boolean;
+  threadRepliesEnabledProp?: boolean;
+  typingEventsEnabledProp?: boolean;
+  uploadsEnabledProp?: boolean;
+}) => {
+  const ChannelCapabilitiesContext: ChannelCapabilitiesContextValue = useMemo(() => {
+    const capabilities = (channel?.data?.own_capabilities || []) as Array<string>;
+
+    const channelCapabilitiesContext = {
+      banChannelMembers: capabilities.includes('ban-channel-members') ?? true,
+      deleteAnyMessage: capabilities.includes('delete-any-message') ?? true,
+      deleteOwnMessage: capabilities.includes('delete-own-message') ?? true,
+      flagMessage: capabilities.includes('flag-message') ?? true,
+      muteChannel: mutesEnabledProp ?? capabilities.includes('mute-channel') ?? true,
+      muteUser: capabilities.includes('mute-user') ?? true,
+      pinMessage: pinMessageEnabledProp ?? capabilities.includes('pin-message') ?? true,
+      quoteMessage: quotedRepliesEnabledProp ?? capabilities.includes('quote-message') ?? true,
+      readEvents: readEventsEnabledProp ?? capabilities.includes('read-events') ?? true,
+      sendMessage: capabilities.includes('send-message') ?? true,
+      sendReaction: reactionsEnabledProp ?? capabilities.includes('send-reaction') ?? true,
+      sendReply: threadRepliesEnabledProp ?? capabilities.includes('send-reply') ?? true,
+      typingEvents: typingEventsEnabledProp ?? capabilities.includes('typing-events') ?? true,
+      updateAnyMessage: capabilities.includes('update-any-message') ?? true,
+      updateOwnMessage: capabilities.includes('update-own-message') ?? true,
+      uploadFile: uploadsEnabledProp ?? capabilities.includes('upload-file') ?? true,
+    };
+
+    return channelCapabilitiesContext;
+  }, [
+    channel.id,
+    mutesEnabledProp,
+    pinMessageEnabledProp,
+    quotedRepliesEnabledProp,
+    readEventsEnabledProp,
+    reactionsEnabledProp,
+    threadRepliesEnabledProp,
+    typingEventsEnabledProp,
+    uploadsEnabledProp,
+  ]);
+
+  return ChannelCapabilitiesContext;
+};
```

Comment:
I tried to map out a bit how this worked, and if we could avoid some of the duplication - I don't think this necessarily is better, but it handles mapping from our chosen keys in one place, which capabilities can be overridden in one place, and keeps (currently) the capabilities set on the channel itself to be the source of truth.

Sharing it in case you think an iteration along these lines could make sense, but if you don't I'm not married to it :D Just an attempt in the direction of being able to skim it a bit.

I moved the capabilities that can be overridden into it's own key, it should be easy to switch those details around.

(( This is all to say, this is pretty much notes I've made myself, and you can take it or leave it and I'm not too attached either way :D ))


```typescript
const allCapabilities = {
    banChannelMembers: 'ban-channel-members',
    deleteAnyMessage: 'delete-any-message',
    deleteOwnMessage: 'delete-own-message',
    flagMessage: 'flag-message',
    muteChannel: 'mute-channel',
    muteUser: 'mute-user',
    pinMessage: 'pin-message',
    quoteMessage: 'quote-message',
    readEvents: 'read-events',
    sendMessage: 'send-message',
    sendReaction: 'send-reaction',
    sendReply: 'send-reply',
    typingEvents: 'typing-events',
    updateAnyMessage: 'update-any-message',
    updateOwnMessage: 'update-own-message',
    uploadFile: 'upload-file',
}

type Capability = keyof typeof capabilities;

const overridableCapabilities: Record<string, Capability> = [
    mute: "muteUser",
    pinMessage: "pinMessage",
    quotedReplies: "quoteMessage",
    reactions: "sendReaction",
    readEvents: "readEvents",
    threadReplies: "sendReply",
    typingEvents: "typingEvents",
    uploads: "uploadFile",
]

type OverridableCapabilities = Record<keyof typeof overridableCapabilities, boolean>;
type ChannelCapabilities = Record<Capability, boolean>;

const normalizeCapabilityKey = (s: keyof typeof overridableCapabilities): Capability =>
    overridableCapabilities.hasOwnProperty(s)
    ? overridableCapabilities[s]
    : throw new Error(`The channel capacity ${s} is not an overridable capacity`)

const mapAllChannelCapabilities = (channelCapabilities: string[]): ChannelCapabilities =>
    Object.keys(allCapabilities).reduce((result, capability) => ({
        ...result,
        [capability]: !!channelCapabilities.includes(allCapabilities[capability]),
    }): {});

const mapAllOverriddenCapabilities = (capabilities: OverridableCapabilities): ChannelCapabilities =>
    Object.entries(capabilities).reduce((result, capability) => {
            const [key, enabled] = capability;
            return {...r, [normalizeCapabilityKey(key)]: enabled };
    }, {});

export const useCreateChannelCapabilitiesContext = <
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType,
>({
  channel,
  capabilities,
}: {
    channel: Channel<At, Ch, Co, Ev, Me, Re, Us>;
    capabilities: OverridableCapabilities;
}) => {
    const ChannelCapabilitiesContext: ChannelCapabilitiesContextValue = useMemo(() => {
        const channelCapabilities = (channel?.data?.own_capabilities || []) as Array<string>;
        

        return {
            ...mapAllChannelCapabilities(channel),
            ...mapAllOverriddenCapabilities(capabilities)
        }
    }, [
        ...capabilities
    ]);

    return ChannelCapabilitiesContext;
}
```

Reply:
(also naming things is hard, so some things might be named a bit funky)